pennylane/pauli/utils.py

```diff
@@ -92,20 +93,100 @@ def is_pauli_word(observable):
     >>> is_pauli_word(4 * qml.PauliX(0) @ qml.PauliZ(0))
     True
     """
+    return False
+
+
+@is_pauli_word.register(PauliX)
+@is_pauli_word.register(PauliY)
+@is_pauli_word.register(PauliZ)
+@is_pauli_word.register(Identity)
+def _is_pw_pauli(
+    observable: Union[PauliX, PauliY, PauliZ, Identity]
+):  # pylint:disable=unused-argument
+    return True
+
+
+@is_pauli_word.register
+def _is_pw_tensor(observable: Tensor):
     pauli_word_names = ["Identity", "PauliX", "PauliY", "PauliZ"]
-    if isinstance(observable, Tensor):
-        return set(observable.name).issubset(pauli_word_names)
+    return set(observable.name).issubset(pauli_word_names)
+
+
+@is_pauli_word.register
+def _is_pw_ham(observable: Hamiltonian):
+    return False if len(observable.ops) != 1 else is_pauli_word(observable.ops[0])
+
+
+@is_pauli_word.register
+def _is_pw_prod(observable: Prod):
+    return all(is_pauli_word(op) for op in observable)
+
+
+@is_pauli_word.register
+def _is_pw_sprod(observable: SProd):
+    return is_pauli_word(observable.base)
+
+
+@singledispatch
+def pauli_word_prefactor(observable):
+    """If the operator provided is a valid Pauli word (i.e a single term which may be a tensor product
+    of pauli operators), then this function extracts the prefactor.
+
+    Args:
+        observable (~.Operator): the operator to be examined
+
+    Returns:
+        Union[int, float, complex]: The scaling/phase coefficient of the Pauliword.
+
+    Raises:
+        ValueError: If an operator is provided that is not a valid Pauli word.
+
+    **Example**
+
+    >>> pauli_word_prefactor(qml.Identity(0))
+    1
+    >>> pauli_word_prefactor(qml.PauliX(0) @ qml.PauliY(1))
+    1
+    >>> pauli_word_prefactor(qml.PauliX(0) @ qml.PauliY(0))
+    1j
+    """
+    raise ValueError(f"Expected a valid Pauli word, got {observable}")
+
+
+@pauli_word_prefactor.register(PauliX)
+@pauli_word_prefactor.register(PauliY)
+@pauli_word_prefactor.register(PauliZ)
+@pauli_word_prefactor.register(Identity)
+def _pw_prefactor_pauli(
+    observable: Union[PauliX, PauliY, PauliZ, Identity]
+):  # pylint:disable=unused-argument
+    return 1
+
+
+@pauli_word_prefactor.register
+def _pw_prefactor_tensor(observable: Tensor):
+    if is_pauli_word(observable):
+        from .conversion import pauli_sentence
```

Comment:
Any thoughts on this? I don't think I can import this top level since `pauli/conversion.py` uses some functionality from this `utils.py` file. Should I just silence the warning?

Reply:
Or we could move these to a separate submodule that depends on pauli arithemtic.  That way we can avoid circular dependencies.